code/modules/mob/living/silicon/mommi/life.dm

```diff
@@ -263,4 +265,18 @@
 	canmove = !(paralysis || stunned || knockdown || locked_to || lockcharge || anchored)
 	return canmove
 
+/mob/living/silicon/robot/mommi/proc/check_locked_zlevel()
+	if(!locked_to_z)
+		return
+
+	var/datum/zLevel/current_zlevel = get_z_level(src)
+	if(!current_zlevel)
+		return
+	if(current_zlevel.z != locked_to_z)
+		to_chat(src, "<span class='userdanger'>Your hardware detects that you have left your intended location. Initiating self-destruct.</span>")
+		if(mmi) //no sneaking brains away
```

Comment:
You could use dust() which is functionally like gib() except the mmi is qdeled for you so it saves you two lines of copy paste here